robot/web/web.go

```diff
@@ -891,6 +895,36 @@ func (svc *webService) initMux(options weboptions.Options) (*goji.Mux, error) {
 		mux.HandleFunc(pat.New("/debug/pprof/trace"), pprof.Trace)
 	}
 
+	// serve resource graph visualization
+	// TODO: hide behind option
+	mux.HandleFunc(pat.New("/debug/graph"), func(w http.ResponseWriter, r *http.Request) {
+		localRobot, isLocal := svc.r.(robot.LocalRobot)
+		if !isLocal {
+			return
+		}
+		dot, err := localRobot.Dot()
+		if err != nil {
+			return
+		}
+		gv := graphviz.New()
+		defer gv.Close()
+
+		graph, err := graphviz.ParseBytes(dot)
```

Comment:
Neat library! I see this expects a byte array. Which I find odd? I don't think you need to change anything here, but because I thought about the options, I'll write down my reaction for posterity.

I probably would have kept the "export"/"dot" function returning a string and just have this call do the type conversion to a utf-8 byte array.

The reason being, if we later found out this library expected utf-16 encoded bytes (perhaps some non-ascii characters in resource names could surface this), we'd have to change the `[]byte(sb.String())` in the export method. Granted, any non-utf-8 example is admittedly a bit contrived. But philosophically I'd argue the choice of how to encode a string into bytes for a library call should be delayed until making use of that method.

Reply:
great points - I'll keep the keep the return signature as `string`